lib/ansible/modules/network/netvisor/pn_port_cos_bw.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/python
+# Copyright: (c) 2018, Pluribus Networks
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: pn_port_cos_bw
+author: "Pluribus Networks (@rajaspachipulusu17)"
+version_added: "2.8"
+short_description: CLI command to modify port-cos-bw
+description:
+  - This module can be used to update bw settings for CoS queues.
+options:
+  pn_cliswitch:
+    description:
+      - Target switch to run the CLI on.
+    required: False
+    type: str
+  state:
+    description:
+      - State the action to perform. Use C(update) to modify the port-cos-bw.
+    required: True
+    type: str
+    choices: ['update']
+  pn_max_bw_limit:
+    description:
+      - Maximum b/w in percentage.
+    required: False
+    type: str
+  pn_cos:
+    description:
+      - CoS priority.
+    required: False
+    type: str
+  pn_port:
+    description:
+      - physical port number.
+    required: False
+    type: str
+  pn_weight:
+    description:
+      - Scheduling weight (1 to 127) after b/w guarantee met.
+    required: False
+    type: str
+    choices: ['priority', 'no-priority']
+  pn_min_bw_guarantee:
+    description:
+      - Minimum b/w in precentage.
+    required: False
+    type: str
+"""
+
+EXAMPLES = """
+- name: port cos bw modify
+  pn_port_cos_bw:
+    pn_cliswitch: "sw01"
+    state: "update"
+    pn_port: "1"
+    pn_cos: "0"
+    pn_min_bw_guarantee: "60"
+
+- name: port cos bw modify
+  pn_port_cos_bw:
+    pn_cliswitch: "sw01"
+    state: "update"
+    pn_port: "all"
+    pn_cos: "0"
+"""
+
+RETURN = """
+command:
+  description: the CLI command run on the target node.
+  returned: always
+  type: str
+stdout:
+  description: set of responses from the port-cos-bw command.
+  returned: always
+  type: list
+stderr:
+  description: set of error responses from the port-cos-bw command.
+  returned: on error
+  type: list
+changed:
+  description: indicates whether the CLI caused changes on the target.
+  returned: always
+  type: bool
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.netvisor.pn_nvos import pn_cli, run_cli
+
+
+def main():
+    """ This section is for arguments parsing """
+
+    state_map = dict(
+        update='port-cos-bw-modify'
+    )
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            pn_cliswitch=dict(required=False, type='str'),
```

Comment:
`required=False` is the default value and not required to be passed explicitly. Applies to other places as well.

Reply:
not required to be passed explicitly. we can either pass it or we may not pass it. Depends on requirement.